### PR TITLE
Prevent calling LICM on removed loops

### PR DESCRIPTION
**Pull request by krzysz00**
Hopefully fix the crashes-after-a-while issues that hit CI.

Also add a missing #include to ArithToAMDGPU

### CODE REVIEW

File: external/llvm-project/mlir/include/mlir/Conversion/ArithToAMDGPU/ArithToAMDGPU.h [hunk 1]
```diff
@@ -10,6 +10,7 @@
 #define MLIR_CONVERSION_ARITHTOAMDGPU_ARITHTOAMDGPU_H
 
 #include <memory>
+#include <string>
```
Comment (sjw36) [inline external/llvm-project/mlir/include/mlir/Conversion/ArithToAMDGPU/ArithToAMDGPU.h:13]: is this needed?

**Review by sjw36 — APPROVED** [review]
Yes, looks good.

File: external/llvm-project/mlir/include/mlir/Conversion/ArithToAMDGPU/ArithToAMDGPU.h (hunk 1 above)
Comment (krzysz00) [reply external/llvm-project/mlir/include/mlir/Conversion/ArithToAMDGPU/ArithToAMDGPU.h:13]: Yeah, doesn't build on my machine without it, and Giuseppe's got a separate PR for that line alone (thought I'd fix it here)

**Review by giuseros — APPROVED** [review]
LGTM!

**Pull request merged**